### PR TITLE
Update WayDataSource

### PR DESCRIPTION
**Pull request by at-hoavo**
#### Related issues (Required)
- Task #
- Bug #

#### Description (Required)
- [x] Update WayDataSource

### CODE REVIEW

File: app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt [hunk 1]
```diff
@@ -1,12 +1,38 @@
...
  * Copyright © 2017 AsianTech inc.
  * Created by vinh.huynh on 11/24/17.
  */
 interface WayDataSource {
-    /**
-     * Function get data
-     */
-    fun getWayData()
+
+    fun createUser(userParams:UserParams)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt [hunk 2]
```diff
@@ -1,12 +1,38 @@
...
+
+    fun updateUser(userParams:UserParams)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt [hunk 3]
```diff
@@ -1,12 +1,38 @@
...
+
+    fun getAddressLocation(latLng: String): Observable<LocationResponse>
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt [hunk 4]
```diff
@@ -1,12 +1,38 @@
...
+
+    fun getLocationDetail(placeId: String?, key: String): Observable<ResultPlaceDetail>
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt [hunk 5]
```diff
@@ -1,12 +1,38 @@
...
+
+    fun searchLocations(input: String,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt [hunk 6]
```diff
@@ -1,12 +1,38 @@
...
+                        key: String,
+                        language: String = "vi",
+                        sensor: Boolean = false): Observable<AutoCompleteResult>
+
+    fun createGroup(name: String): Observable<Group>
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt [hunk 7]
```diff
@@ -1,12 +1,38 @@
...
+
+    fun getGroupInfo(groupId: String): Observable<Group>
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt [hunk 8]
```diff
@@ -1,12 +1,38 @@
...
+
+    fun getMembersList(groupId: String): Observable<UserListResult>
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt [hunk 9]
```diff
@@ -1,12 +1,38 @@
...
+
+    fun addUserToGroup(userId: String, body: BodyAddUserToGroup): Observable<User>
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt [hunk 10]
```diff
@@ -1,12 +1,38 @@
...
+
+    fun removeUserFromGroup(userId: String, body: BodyAddUserToGroup): Observable<User>
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt (hunk 1 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt (hunk 2 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/data/source/remote/WayRemoteDataSource.kt [hunk 11]
```diff
@@ -1,13 +1,55 @@
...
+
+    override fun getAddressLocation(latLng: String): Observable<LocationResponse> {
+        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+    }
+
+    override fun getLocationDetail(placeId: String?, key: String): Observable<ResultPlaceDetail> {
+        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+    }
+
+    override fun searchLocations(input: String, key: String, language: String, sensor: Boolean): Observable<AutoCompleteResult> {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/remote/WayRemoteDataSource.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt [hunk 12]
```diff
@@ -1,12 +1,39 @@
...
  * Created by vinh.huynh on 11/24/17.
  */
-interface WayDataSource {
-    /**
-     * Function get data
-     */
-    fun getWayData()
+internal interface WayDataSource {
+
+    fun createUser(userParams:UserParams)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt [hunk 13]
```diff
@@ -1,12 +1,39 @@
...
+
+    fun updateUser(userParams:UserParams)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt [hunk 14]
```diff
@@ -1,12 +1,39 @@
...
+
+    fun getUser():Observable<User>
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt [hunk 15]
```diff
@@ -1,12 +1,39 @@
...
+
+    fun getAddressLocation(latLng: String): Observable<LocationResponse>
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt [hunk 16]
```diff
@@ -1,12 +1,39 @@
...
+
+    fun getLocationDetail(placeId: String?, key: String): Observable<ResultPlaceDetail>
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt (hunk 12 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt (hunk 13 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt (hunk 14 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt [hunk 17]
```diff
@@ -1,12 +1,105 @@
...
  */
-interface WayDataSource {
+internal interface WayDataSource {
+
+    /**
+     *  This method to create new User.
+     *
+     *  @param userParams contain user information for Hypertrack to create new user.
+     */
+    fun createUser(userParams:UserParams)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt [hunk 18]
```diff
@@ -1,12 +1,105 @@
...
+
+    /**
+     *  This method to update User.
+     *
+     *  @param userParams: contain user information for Hypertrack to update.
+     */
+    fun updateUser(userParams:UserParams)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt [hunk 19]
```diff
@@ -1,12 +1,105 @@
...
+
+    /**
+     *  Get information user.
+     *
+     *  @return User object which is Observable.
+     */
+    fun getUser():Observable<User>
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/WayDataSource.kt]: WARNING:

**Comment by at-hoavo** [comment]
@at-tienhoang please check for me, thank you!

File: app/src/main/java/vn/asiantech/way/data/source/remote/WayRemoteDataSource.kt [hunk 20]
```diff
@@ -1,13 +1,58 @@
...
+
+    override fun getAddressLocation(latLng: String): Observable<MutableList<LocationAddress>> {
+        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+    }
+
+    override fun getLocationDetail(placeId: String?, key: String): Observable<ResultPlaceDetail> {
+        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+    }
+
+    override fun searchLocations(input: String, key: String, language: String, sensor: Boolean): Observable<AutoCompleteResult> {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/remote/WayRemoteDataSource.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/data/source/remote/response/ResponseStatus.kt [hunk 21]
```diff
@@ -0,0 +1,9 @@
+package vn.asiantech.way.data.source.remote.response
+
+/**
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/remote/response/ResponseStatus.kt]: INFO: Empty block of code detected. As they serve no purpose they should be removed.

File: app/src/main/java/vn/asiantech/way/data/source/remote/response/ResponseStatus.kt [hunk 22]
```diff
@@ -0,0 +1,9 @@
...
+ *  Copyright © 2017 AsianTech inc.
+ *  Created by hoavot on 28/11/2017.
+ */
+data class ResponseStatus(val status: Boolean, val message: String = "") {
+
+}
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/remote/response/ResponseStatus.kt]: WARNING: Checks whether files end with a line separator.

**Pull request merged**